Mathlib/Algebra/Field/IsField.lean

```diff
@@ -63,10 +63,13 @@ noncomputable def IsField.toSemifield {R : Type u} [Semiring R] (h : IsField R)
   inv_zero := dif_pos rfl
   mul_inv_cancel a ha := by convert Classical.choose_spec (h.mul_inv_cancel ha); exact dif_neg ha
   nnqsmul := _
+  nnqsmul_def q a := rfl
 
 /-- Transferring from `IsField` to `Field`. -/
 noncomputable def IsField.toField {R : Type u} [Ring R] (h : IsField R) : Field R :=
-  { ‹Ring R›, IsField.toSemifield h with qsmul := _ }
+  { ‹Ring R›, IsField.toSemifield h with
```

Comment:
Just flagging for the other reviewers: I don't have an opinion on these - I trust you these are necessary later.

Reply:
This change is harmless, but I don't understand why it is needed.

Clearly the `_` is being populated by elaborating the `rfl` proof, but why is this impacted by whether the `rfl` proof is inlined or inserted automatically behind the scenes?

Reply:
Under `byAsSorry`, `rfl` will never even run.

Reply:
Oh, is the difference here `by rfl` vs `rfl`?